lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -70,19 +70,26 @@ defmodule Mix.Tasks.Compile.Elixir do
   @doc """
   Returns Elixir manifests.
   """
-  def manifests, do: [manifest]
-  defp manifest, do: Path.join(Mix.Project.manifest_path, @manifest)
+  def manifests, do: Map.values(manifest_map)
+  defp manifest_map do
+    Enum.map(sources, fn (source) ->
+      {source, Path.join(Mix.Project.manifest_path, @manifest_root <> "." <> String.replace(source, "/", "_"))}
```

Comment:
`String.replace` here is to sanitize file names. I'm sure there's a cleaner approach.